src/sentry/api/endpoints/source_map_debug.py

```diff
@@ -191,9 +189,92 @@ def _find_matches(self, release_artifacts, unified_path):
+    def _extract_release(self, event, project):
+        release_version = event.get_tag("sentry:release")
+
+        if not release_version:
+            return self._create_response(issue=SourceMapProcessingIssue.MISSING_RELEASE)
+        try:
+            release = Release.objects.get(
+                organization=project.organization, version=release_version
+            )
+        except Release.DoesNotExist:
+            return self._create_response(issue=SourceMapProcessingIssue.MISSING_RELEASE)
+
+        return release
+
+    def _verify_dist_matches(self, release, event, artifact, filename):
+        try:
+            dist = Distribution.objects.get(release=release, name=event.dist)
+        except Distribution.DoesNotExist:
+            return self._create_response(
+                issue=SourceMapProcessingIssue.DIST_MISMATCH,
+                data={"eventDist": event.dist, "fileName": filename},
+            )
+
+        if artifact.dist_id != dist.id:
+            return self._create_response(
+                issue=SourceMapProcessingIssue.DIST_MISMATCH,
+                data={"eventDist": dist.id, "artifactDist": artifact.dist_id, "fileName": filename},
+            )
+        return dist
+
+    def _find_matching_artifact(self, release_artifacts, urlparts, abs_path, filename):
+        unified_path = self._unify_url(urlparts)
+        full_matches, partial_matches = self._find_matches(release_artifacts, unified_path)
+
+        artifact_names = [artifact.name for artifact in release_artifacts]
+
+        if len(full_matches) == 0:
+            if len(partial_matches) > 0:
+                partial_match = partial_matches[0]
+                url_prefix = self._find_url_prefix(urlparts.path, partial_match.name)
+
+                return self._create_response(
+                    issue=SourceMapProcessingIssue.PARTIAL_MATCH,
+                    data={
+                        "absPath": abs_path,
+                        "partialMatchPath": partial_match.name,
+                        "fileName": filename,
+                        "unifiedPath": unified_path,
+                        "urlPrefix": url_prefix,
+                        "artifactNames": artifact_names,
+                    },
+                )
+            return self._create_response(
+                issue=SourceMapProcessingIssue.NO_URL_MATCH,
+                data={
+                    "absPath": abs_path,
+                    "fileName": filename,
+                    "unifiedPath": unified_path,
+                    "artifactNames": artifact_names,
+                },
+            )
+        return full_matches[0]
+
+    def _get_filename(self, event, exception_idx, frame_idx):
+        exceptions = event.interfaces["exception"].values
+        frame_list = exceptions[int(exception_idx)].stacktrace.frames
+        frame = frame_list[int(frame_idx)]
+        filename = frame.filename
+        return filename
+
+    def _discover_sourcemap_url(self, artifact):
```

Comment:
Maybe you can somehow extract and/or reuse `discover_sourcemap` from JS processor?

Reply:
I don't think we are able to re use `discover_sourcemap` so my code was based off it. ill try to keep it as similar as possible